src/test/java/org/elasticsearch/index/translog/fs/AbstractTranslogTests.java

```diff
@@ -664,6 +613,170 @@ public void testVerifyTranslogIsNotDeleted() throws IOException {
             translog.close();
         }
 
-        assertTrue(Files.exists(path.resolve("translog-1")));
+        assertFileIsPresent(translog, 1);
+    }
+
+    @Test
+    public void testConcurrentWriteViewsAndSnapshot() throws Throwable {
```

Comment:
complex test is very complex, needs looooots of comments so others can understand it when they are debugging :)